tests/filecheck/dialects/scf/for_args_types.mlir

```diff
@@ -1,14 +1,26 @@
-// RUN: xdsl-opt %s --verify-diagnostics | filecheck %s
+// RUN: xdsl-opt %s --verify-diagnostics --split-input-file | filecheck %s
+
 "builtin.module"() ({
-  %lb = "arith.constant"() {"value" = 0 : i32} : () -> index
-  %ub = "arith.constant"() {"value" = 42 : index} : () -> index
-  %step = "arith.constant"() {"value" = 7 : index} : () -> index
-  %lbi = "arith.constant"() {"value" = 0 : i32} : () -> i32
-  %ubi = "arith.constant"() {"value" = 42 : index} : () -> i32
-  %stepi = "arith.constant"() {"value" = 7 : index} : () -> i32
-// CHECK: i32 should be of base attribute index
+  %lbi = "test.op"() {"value" = 0 : index} : () -> !test.type<"int">
+  %ub = "test.op"() {"value" = 42 : index} : () -> index
+  %step = "test.op"() {"value" = 7 : index} : () -> index
```

Comment:
the TestOp has a variadic number of results, so you only need one of them that produces three SSAValues. Also, the value attribute is now useless and can be removed.